lte/gateway/c/core/oai/tasks/ngap/ngap_state_manager.cpp

```diff
@@ -93,6 +97,64 @@ void NgapStateManager::create_state() {
   create_ngap_imsi_map();
 }
 
+status_code_e NgapStateManager::read_state_from_db() {
+#if !MME_UNIT_TEST
+  /* Data store is Redis db. In this case actual call is made to Redis db */
+  StateManager::read_state_from_db();
+#else
+  /* Data store is a map defined in NGAPClientServicer.In this case call is NOT
+   * made to Redis db */
+  if (persist_state_enabled) {
+    S1apState state_proto = S1apState();
+    std::string proto_str;
+    // Reads from the NGAPClientServicer DataStore Map
+    // (map_ngapState_tableKey_protoStr)
+    if (NGAPClientServicer::getInstance().map_ngapState_tableKey_protoStr.get(
+            table_key, &proto_str) != magma::MAP_OK) {
+      OAILOG_DEBUG(LOG_MME_APP, "Failed to read proto from db \n");
+      return RETURNerror;
+    }
+    // Deserialization Step
+    if (!state_proto.ParseFromString(proto_str)) {
+      return RETURNerror;
+    }
+    NgapStateConverter::proto_to_state(state_proto, state_cache_p);
+  }
+#endif
+  return RETURNok;
+}
+
+void NgapStateManager::write_state_to_db() {
+#if !MME_UNIT_TEST
+  /* Data store is Redis db. In this case actual call is made to Redis db */
+  StateManager::write_state_to_db();
+#else
+  /* Data store is a map defined in NGAPClientServicer.In this case call is NOT
+   * made to Redis db */
+  if (persist_state_enabled) {
```

Comment:
same question here as above

Reply:
Removed the flag. Addressed the review comment